ChatApplication/Controllers/MessagesController.cs

```diff
@@ -20,59 +22,33 @@ namespace ChatApplication.Controllers
     //[Authorize]
     public class MessagesController : Controller
     {
-        private readonly IDistributedCache _distributedCache;
         private readonly ApplicationDbContext _context;
         private readonly UserManager<AppUser> _userManager;
+        private readonly ICacheExtensionsService _cacheExtensionsService;
 
+        string cacheKey = "Allmessages0";
+
+        public MessagesController(ApplicationDbContext context, UserManager<AppUser> userManager, ICacheExtensionsService cacheExtensionsService)
```

Comment:
we discussed to create 1 service which will have db and redid inside and control the interactions between those. the controller should be aware of only that 1 service. Remove db and redid from controller

Reply:
Create IMessagesService and manage db/redis interactions from there